src/sources/socket/mod.rs

```diff
@@ -247,6 +247,13 @@ impl SourceConfig for SocketConfig {
                         LegacyKey::InsertIfEmpty,
                     );
 
+                let tls_client_metadata_path = config
+                    .tls()
+                    .as_ref()
+                    .and_then(|tls| tls.client_metadata_key.as_ref())
+                    .and_then(|x| parse_value_path(x).ok())
+                    .map(LegacyKey::InsertIfEmpty);
```

Comment:
Based on the usage looks like this should be 
```suggestion
                    .map(LegacyKey::Overwrite);
```

Reply:
Good catch. Thanks.